src/urllib3/connectionpool.py

```diff
@@ -922,12 +945,12 @@ def _prepare_proxy(self, conn):
 
         conn.set_tunnel(self._proxy_host, self.port, self.proxy_headers)
 
-        if self.proxy.scheme == "https":
+        if self.proxy and self.proxy.scheme == "https":
             conn.tls_in_tls_required = True
 
         conn.connect()
 
-    def _new_conn(self):
+    def _new_conn(self) -> ClientHTTPSConnection:  # type: ignore
```

Comment:
Same as above, why are we using `ClientHTTPSConnection`?

Reply:
we have `Return a fresh :class:`http.client.HTTPSConnection`.` in docstring.
Isn't it right?